src/hotspot/share/opto/ifnode.cpp

```diff
@@ -837,8 +857,94 @@ bool IfNode::has_only_uncommon_traps(ProjNode* proj, ProjNode*& success, ProjNod
   return false;
 }
 
+// There might be an AddINode (marked with *) with a constant increment
```

Comment:
Never found a SubNode in this case. Always AddI node with integer constant as second input (this constant 
can be positive or negative values)

Reply:
That's because we always convert `SubINodes` with constant second operand to `AddINodes`:
https://github.com/openjdk/jdk/blob/3d3eb5c8d31ecd80309fdcbb67cd2aeeb2037c52/src/hotspot/share/opto/subnode.cpp#L181

Reply:
Okay I see, thanks for the explanation.